core/cache/tables.go

```diff
@@ -17,148 +17,110 @@
-	rwmutex   sync.RWMutex
-	capacity  int
-}
-
-// GetInstance returns SFTables singleton instance
-func GetInstance() *SFTables {
-	once.Do(func() {
-		instance = newSFTables(cacheSize)
-	})
-	return instance
+	contTable  map[string]*sfgo.Container
+	procTable  map[uint64][]*sfgo.Process
+	fileTable  map[uint64]*sfgo.File
+	ptreeTable map[uint64][]*sfgo.Process
 }
 
-// newSFTables creates a new SFTables instance.
-func newSFTables(capacity int) *SFTables {
+// NewSFTables creates a new SFTables instance.
+func NewSFTables() *SFTables {
 	t := new(SFTables)
-	if capacity < 1 {
-		logger.Error.Println("Cache capacity must be greater than 1")
-		return nil
-	}
-	t.capacity = capacity
-	t.contTable = cqueue.NewFIFO()
-	t.procTable = cqueue.NewFIFO()
-	t.fileTable = cqueue.NewFIFO()
-	t.contTable.Enqueue(cmap.New())
-	t.procTable.Enqueue(cmap.New())
-	t.fileTable.Enqueue(cmap.New())
+	t.new()
 	return t
 }
 
-// Reset pushes a new set of empty maps into the cache.
-func (t *SFTables) Reset() {
-	t.rwmutex.Lock()
-	defer t.rwmutex.Unlock()
-	t.reset(t.contTable)
-	t.reset(t.procTable)
-	t.reset(t.fileTable)
+func (t *SFTables) new() {
+	t.contTable = make(map[string]*sfgo.Container)
+	t.procTable = make(map[uint64][]*sfgo.Process)
+	t.fileTable = make(map[uint64]*sfgo.File)
+	t.ptreeTable = make(map[uint64][]*sfgo.Process)
 }
 
-func (t *SFTables) reset(queue *cqueue.FIFO) {
-	queue.Enqueue(cmap.New())
-	if queue.GetLen() > t.capacity {
-		queue.Remove(0)
-	}
+// Reset pushes a new set of empty maps into the cache.
+func (t *SFTables) Reset() {
+	t.new()
 }
 
 // GetCont retrieves a cached container object by ID.
-func (t *SFTables) GetCont(ID string) *sfgo.Container {
-	t.rwmutex.RLock()
-	defer t.rwmutex.RUnlock()
-	for i := 0; i < t.contTable.GetLen(); i++ {
-		m, _ := t.contTable.Get(i)
-		table := m.(cmap.ConcurrentMap)
-		if v, ok := table.Get(ID); ok {
-			return v.(*sfgo.Container)
-		}
-	}
-	return nil
+func (t *SFTables) GetCont(ID string) (co *sfgo.Container) {
+	co = t.contTable[ID]
+	return
 }
 
 // SetCont stores a container object in the cache.
 func (t *SFTables) SetCont(ID string, o *sfgo.Container) {
-	t.rwmutex.RLock()
-	m, _ := t.contTable.Get(t.contTable.GetLen() - 1)
-	t.rwmutex.RUnlock()
-	table := m.(cmap.ConcurrentMap)
-	table.Set(ID, o)
+	t.contTable[ID] = o
 }
 
 // GetProc retrieves a cached process object by ID.
-func (t *SFTables) GetProc(ID sfgo.OID) *sfgo.Process {
-	t.rwmutex.RLock()
-	defer t.rwmutex.RUnlock()
-	for i := 0; i < t.procTable.GetLen(); i++ {
-		m, _ := t.procTable.Get(i)
-		table := m.(cmap.ConcurrentMap)
-		if v, ok := table.Get(t.getHash(ID)); ok {
-			return v.(*sfgo.Process)
+func (t *SFTables) GetProc(ID sfgo.OID) (po *sfgo.Process) {
+	if p, ok := t.procTable[hash.GetHash(ID)]; ok {
+		if v := p[sfgo.SFObjectStateMODIFIED]; v != nil {
+			po = v
+		} else if v := p[sfgo.SFObjectStateCREATED]; v != nil {
+			po = v
+		} else if v := p[sfgo.SFObjectStateREUP]; v != nil {
+			po = v
```

Comment:
Do we still need to store processes by their state objects given we are always passing the latest contextual objects with an event or flow?